doc/http_methods.rst

```diff
@@ -0,0 +1,221 @@
+        self._queryable = QueryableEntities()
+        self._metadata_string = metadata
+        self._restrictions = restrictions
+        Config.init()
+        Config.fuzzer.http_method_enabled = method
+
+In config.py, this would be used to build the Config object, which would be looked up during query construction and check which HTTP method is set.
+
+Truncating Query Options
+------------------------
+
+For Odata queries other than **GET**, query options need to be truncated. To implement this, in fuzzer.py *Query.build_string()* would first check if *Config.fuzzer.http_method_enabled == "GET"* before generating the options and appending them. 
+The list of options are illustrated in the table above.
+
+
+Generating a Body
+-----------------
+
+For PUT and POST queries, a new element for the queries need to be generated i.e. the **body** of the request. 
+The first change is returning a tuple of *query, body* instead of just the *query* from *SingleQueryable.generate()*.
+A new function *generate_body()* is added which fetches the proprties from the metadata and calls the generator on each of them, and appends them is a dictionary.
+The dictionary is jsonified before being returned as the body. This process is skipped for **GET** and **DELETE**,and they return an empty jsonified string instead, to be compatible with the tuple returned. The changes made to the generators are described further below.
+
+
+Differentiating between PUT and POST
+------------------------------------
+
+PUT is idempotent and address single entities, whereas POST isnt idempotent and addresses multiple entities. Therefore changes are made in *DirectBuilder._append_queryable()* so that PUT avoids generating multiple entities and POST avoids generating single entities, during query generation.
+
+.. code-block:: python
+
+    def _append_queryable(self, query_group_data):
+        # TODO REFACTOR DRY this method is direct copypaste from DispatchedBuilder just to have a prototype for integration. Intentionally no abstract class at the moment.
+        if Config.fuzzer.http_method_enabled != "POST":
+            self._append_corresponding_queryable(QueryGroupSingle(query_group_data))
+        if Config.fuzzer.http_method_enabled != "PUT":
+            self._append_corresponding_queryable(QueryGroupMultiple(query_group_data))
+            self._append_associated_queryables(query_group_data)
+
+
+Alternative EDM Generators for Body
+-----------------------------------
+
+Some EDM data types have different representation format in the body than in the URI. So the generators needed to adapt for the body implementation. The *generate()* function in the generator classes now have an additional parameter **format** which is provided the value "body". The default generation is done by **generate(format="uri")**. Following is an example of calling generator on a property for body format.
```

Comment:
It would be great to provide some examples of how these different representations look like. I have no idea what representations you are referring to.

Reply:
yeah, actually unit tests are great fist examples of how to use it and what are the real differences .)

Reply:
@AlapanGhosh Please do not forget this in the next PR.